bin/plugin/group-aclkeeper/groupSetServers

```diff
@@ -0,0 +1,190 @@
+#! /usr/bin/env perl
+# vim: set filetype=perl ts=4 sw=4 sts=4 et:
+use common::sense;
+use JSON;
+
+use File::Basename;
+use lib dirname(__FILE__) . '/../../../lib/perl';
+use OVH::Result;
+use OVH::Bastion;
+use OVH::Bastion::Plugin qw( :DEFAULT help );
+use OVH::Bastion::Plugin::ACL;
+
+my $remainingOptions = OVH::Bastion::Plugin::begin(
+    argv               => \@ARGV,
+    header             => "replace a group's current ACL by a new one",
+    userAllowWildcards => 1,
+    options            => {
+        "group=s"     => \my $group,
+        "dry-run"     => \my $dryRun,
+        "skip-errors" => \my $skipErrors,
+    },
+    helptext => <<'EOF',
+Replace a group's current ACL by a new list
+
+Usage: --osh SCRIPT_NAME --group GROUP [OPTIONS]
+
+  --group GROUP  Specify which group this machine should be added to
+  --dry-run      Don't actually modify the ACL, just report whether the input contains errors
+  --skip-errors  Don't abort on STDIN parsing errors, just skip the non-parseable lines
+
+The list of the assets to constitute the new ACL should then be given on STDIN,
+in the following format: ``[USER@]HOST[:PORT][ COMMENT]``, with USER and PORT being optional,
+and HOST being either a hostname, an IP, or an IP block in CIDR notation. The COMMENT is
+also optional, and may contain spaces.
+
+Examples::
+
+server12.example.org
+logs@server
+host1.example.net:2222 host1 on secondary sshd with alternate port
+root@192.0.2.0/24 production database cluster
+EOF
+);
+
+my $fnret;
+
+if (not $group) {
+    help();
+    osh_exit 'ERR_MISSING_PARAMETER', "Missing mandatory parameter 'group'";
+}
+
+$fnret = OVH::Bastion::is_valid_group_and_existing(group => $group, groupType => "key");
+$fnret or osh_exit($fnret);
+
+# get returned untainted value
+$group = $fnret->value->{'group'};
+my $shortGroup = $fnret->value->{'shortGroup'};
+
+$fnret = OVH::Bastion::is_group_aclkeeper(account => $self, group => $shortGroup, superowner => 1);
+$fnret
+  or osh_exit 'ERR_NOT_GROUP_ACLKEEPER',
+  "Sorry, you must be an aclkeeper of group $shortGroup to be able to add servers to it";
+
+osh_info
+  "Specify the entries of the new ACL below, one per line, with the following format: [USER\@]HOST[:PORT][ COMMENT]";
+osh_info "The list ends at EOF (usually CTRL+D).";
+osh_info "You may abort with CTRL+C if needed.";
+
+my @ACL;
+my @errors;
+my $nbLines = 0;
+my $comment;
+while (my $line = <STDIN>) {
+    # trim white spaces
+    $line =~ s/^\s+|\s+$//g;
+
+    # empty line ?
+    $line or next;
+
+    $nbLines++;
+
+    if ($line =~ m{^(?:(\S+)\@)?([a-zA-Z0-9_./-]+)(?::(\d+))?(?:\s+(.+))?$}) {
+        $user    = $1;
```

Comment:
This might be on purpose but using these as they are imported by the plugin mechanism seems odd to me.
I would have expected dedicated declared variables here, to not mix how `$host` et al are used in the plugins

Reply:
This was kinda on purpose, as in the case of this plugin, there is no one single tuple user/host/port, but indeed a list, and it doesn't hurt to reuse these vars.
But if it is confusing, or even just eyebrow-raising, I'll use dedicated vars instead.

Reply:
I find it odd but good enough